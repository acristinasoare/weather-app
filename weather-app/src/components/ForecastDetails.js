function ForecastDetails({ forecast }) {
  const { date, humidity, temperature, wind } = forecast;
  const formattedDate = new Date(date).toDateString();

  return (
    <div>
      <div className="forecast-details" data-testid="forecast-details" />
      <div className="forecast-details__data">{formattedDate}</div>
      <div className="forecast-details__data">Humidity: {humidity} %</div>
      <div className="forecast-details__data">
        Max temperature: {temperature.max}&deg;C
      </div>
      <div className="forecast-details__data">
        Min temperature: {temperature.min}&deg;C
      </div>
      <div className="forecast-details__data">Wind speed: {wind.speed} mph</div>
      <div className="forecast-details__data">
        Wind direction: {wind.direction}
      </div>
    </div>
  );
}

export default ForecastDetails;
