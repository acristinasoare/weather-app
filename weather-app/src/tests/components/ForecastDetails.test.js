import { render } from "@testing-library/react";
import ForecastDetails from "../../components/ForecastDetails";

describe("ForecastDetails", () => {
  const validProps = {
    date: 1111111,
    humidity: 30,
    temperature: {
      min: 20,
      max: 23,
    },
    wind: {
      speed: 15,
      direction: "s",
    },
  };

  it("renders correctly", () => {
    const { asFragment } = render(<ForecastDetails forecast={validProps} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it("renders correct values for props", () => {
    const { getByTestId, getByText } = render(
      <ForecastDetails forecast={validProps} />
    );

    expect(getByTestId("forecast-details")).toBeInTheDocument();
    expect(getByText("Thu Jan 01 1970")).toBeInTheDocument();
    expect(getByText("Humidity: 30 %")).toBeInTheDocument();
    expect(getByText("Min temperature: 20°C")).toBeInTheDocument();
    expect(getByText("Max temperature: 23°C")).toBeInTheDocument();
    expect(getByText("Wind speed: 15 mph")).toBeInTheDocument();
    expect(getByText("Wind direction: s")).toBeInTheDocument();
  });
});
