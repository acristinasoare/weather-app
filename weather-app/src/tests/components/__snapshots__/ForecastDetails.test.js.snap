// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ForecastDetails renders correctly 1`] = `
<DocumentFragment>
  <div>
    <div
      class="forecast-details"
      data-testid="forecast-details"
    />
    <div
      class="forecast-details__data"
    >
      Thu Jan 01 1970
    </div>
    <div
      class="forecast-details__data"
    >
      Humidity: 30 %
    </div>
    <div
      class="forecast-details__data"
    >
      Max temperature: 23°C
    </div>
    <div
      class="forecast-details__data"
    >
      Min temperature: 20°C
    </div>
    <div
      class="forecast-details__data"
    >
      Wind speed: 15 mph
    </div>
    <div
      class="forecast-details__data"
    >
      Wind direction: s
    </div>
  </div>
</DocumentFragment>
`;
